storage-api/src/java/org/apache/hive/common/util/BloomKFilter.java

```diff
@@ -506,18 +505,19 @@ public ElementWrapper(byte[] bytes, int start, int length, int modifiedStart, in
   }
 
   private static class BloomFilterMergeWorker implements Runnable {
-    Queue<ElementWrapper> queue = new LinkedBlockingDeque<>();
+    ArrayBlockingQueue<ElementWrapper> queue;
     private ExecutorService executor;
 
     private byte[] bfAggregation;
     private int bfAggregationStart;
     private int bfAggregationLength;
 
-    public BloomFilterMergeWorker(ExecutorService executor, byte[] bfAggregation, int bfAggregationStart, int bfAggregationLength) {
+    public BloomFilterMergeWorker(ExecutorService executor, int batchSize, byte[] bfAggregation, int bfAggregationStart, int bfAggregationLength) {
       this.executor = executor;
```

Comment:
Do not capture this value.

Reply:
@belugabehr : sorry I cannot get your point, what do you mean?

Reply:
@abstractdog Hey, ya, of course.  This just goes with what I was saying about the relationship between Runnable/Thread/Callable and ExecutorService.  The thread rarely needs a reference to its own ExecutorService.  The only reason this was being capture was to support `!executor.isTerminated()` which is not the correct thing to do.  Please remove the `this.executor` instance variable and/or change the constructor to not include it.

Reply:
sure, I misunderstood, somehow I thought you meant batchSize...yeah, we won't need executor reference if I remove executor.isTerminated() check